src/test/java/com/redhat/pantheon/model/api/SlingResourceTest.java

```diff
@@ -0,0 +1,292 @@
+
+        // Then
+        assertNotNull(model.CHILD.get());
+        assertEquals("prop-value", model.CHILD.get().getResource().getValueMap().get("jcr:property"));
+    }
+
+    @Test
+    public void testIsPresent() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .resource("child",
+                        "jcr:property", "prop-value")
+                .resource("/content/module2",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .commit();
+
+        // When
+        TestResource model1 = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+        TestResource model2 = new TestResource(slingContext.resourceResolver().getResource("/content/module2"));
+
+        // Then
+        assertTrue(model1.CHILD.isPresent());
+        assertFalse(model2.CHILD.isPresent());
+    }
+
+    @Test
+    public void testGetOrCreateChildExists() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .resource("child",
+                        "jcr:property", "prop-value")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        assertNotNull(model.CHILD.getOrCreate());
+        assertEquals("prop-value", model.CHILD.get().getResource().getValueMap().get("jcr:property"));
+    }
+
+    @Test
+    public void testGetOrCreateChildDoesNotExist() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        assertNotNull(model.CHILD.getOrCreate());
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+        // Given
+        ResourceResolver resourceResolver = slingContext.resourceResolver();
+        Resource root = resourceResolver.getResource("/");
+
+        // When
+        TestResource model = new TestResource(
+                resourceResolver.create(root,"test", null));
+        model.NAME.set("my-new-node");
+        model.NUMBER.set(25L);
+        model.commit();
```

Comment:
Say... here's a `model.commit()`. I guess it wasn't important for the earlier test where I asked about it?

Reply:
In this case it is important because as you see I am reloading the model. I am testing that things got stored and re-loaded properly.

Reply:
Makes sense.